Automated Smoketest Build: failed
Number of Error(s): 4
/mnt/disk1/home/vamosax/smoketest/smoketest-7940/HPCC-Platform/system/security/htpasswdSecurity/htpasswdSecurity.cpp:183:5: error: expected ‘;’ before ‘}’ token
make[2]: *** [system/security/htpasswdSecurity/CMakeFiles/htpasswdSecurity.dir/htpasswdSecurity.cpp.o] Error 1
make[1]: *** [system/security/htpasswdSecurity/CMakeFiles/htpasswdSecurity.dir/all] Error 2
make: *** [all] Error 2

--- a/system/security/htpasswdSecurity/htpasswdSecurity.cpp
+++ b/system/security/htpasswdSecurity/htpasswdSecurity.cpp
@@ -179,7 +179,8 @@ private:
 						throw MakeStringException(-1, "htpasswd Password file appears malformed");
 					*colon = (char)NULL;
 					userMap.setValue(next, colon+1);//username, enctypted password
-				} while (next = strtok_r(NULL, seps, &saveptr));
+					next = strtok_r(NULL, seps, &saveptr);
+				} while (next);
 			}
 
 			io->close();

